SYCL/Complex/sycl_complex_stream_test.cpp

```diff
@@ -0,0 +1,93 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %HOST_RUN_PLACEHOLDER %t.out %HOST_CHECK_PLACEHOLDER
+// RUN: %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+// RUN: %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
+// RUN: %ACC_RUN_PLACEHOLDER %t.out %ACC_CHECK_PLACEHOLDER
+
+#include "sycl_complex_helper.hpp"
+
+template <typename T> struct test_sycl_stream_operator {
+  bool operator()(sycl::queue &Q, cmplx<T> init) {
+    auto *cplx_out = sycl::malloc_shared<experimental::complex<T>>(1, Q);
+    cplx_out[0] = experimental::complex<T>(init.re, init.im);
+
+    Q.submit([&](sycl::handler &CGH) {
+      sycl::stream Out(512, 20, CGH);
+      CGH.parallel_for<>(sycl::range<1>(1), [=](sycl::id<1> idx) {
+        Out << cplx_out[idx] << sycl::endl;
+      });
+    });
+
+    return true;
+  }
+};
+
+// Host only tests for std::basic_ostream and std::basic_istream
+template <typename T> struct test_ostream_operator {
+  bool operator()(cmplx<T> init) {
+    experimental::complex<T> c(init.re, init.im);
+
+    std::ostringstream os;
+    os << c;
+
+    std::ostringstream ref_oss;
+    ref_oss << "(" << init.re << "," << init.im << ")";
+
+    if (ref_oss.str() == os.str())
+      return true;
+    return false;
+  }
+};
+
+template <typename T> struct test_istream_operator {
+  bool operator()(cmplx<T> init) {
+    experimental::complex<T> c(init.re, init.im);
+
+    std::ostringstream ref_oss;
+    ref_oss << "(" << init.re << "," << init.im << ")";
+
+    std::istringstream iss(ref_oss.str());
+
+    iss >> c;
+
+    return check_results(c, std::complex<T>(init.re, init.im),
+                         /*is_device*/ false);
+  }
+};
+
+int main() {
+  sycl::queue Q;
+
+  bool test_passes = true;
+
+  // CHECK: (1.5,-1)
+  // CHECK: (1.5,-1)
+  // CHECK: (1.5,-1)
```

Comment:
```suggestion
  // CHECK-COUNT-3: (1.5,-1)
```
Also applies to the following cases.

Reply:
I have updated this check